### PR TITLE
Add cch

### PR DESCRIPTION
**Pull request by crixal**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3120)

### CODE REVIEW

**Review by nikolayilyin — CHANGES_REQUESTED** [review]
nitpicks mostly

File: src/main/scala/beam/router/RoutingWorker.scala
Comment (crixal) [reply src/main/scala/beam/router/RoutingWorker.scala]: I know only one. Fixed

File: src/main/scala/beam/router/RoutingWorker.scala
Comment (crixal) [reply src/main/scala/beam/router/RoutingWorker.scala]: Agree. Renamed `l` to `link`

File: src/main/scala/beam/router/cch/CchWrapper.scala
Comment (crixal) [reply src/main/scala/beam/router/cch/CchWrapper.scala]: It's works fine without tags for nodes.

File: src/main/scala/beam/router/cch/CchWrapper.scala
Comment (JustinPihony) [inline src/main/scala/beam/router/cch/CchWrapper.scala:148]: This should be in parentheses to avoid any issues down the road.

File: src/main/scala/beam/router/cch/CchWrapper.scala
Comment (JustinPihony) [inline src/main/scala/beam/router/cch/CchWrapper.scala:142]: Can the comments be removed?

File: src/main/scala/beam/router/cch/CchWrapper.scala
Comment (JustinPihony) [inline src/main/scala/beam/router/cch/CchWrapper.scala:172]: Could this be more expressive to help users? 

> Windows is not supported for CCH usage, please change `beam.routing.carRouter` to another value.

**Review by JustinPihony — CHANGES_REQUESTED** [review]
A few red flags that need handled before this can be merged.

File: src/main/scala/beam/router/RoutingWorker.scala
Comment (JustinPihony) [inline src/main/scala/beam/router/RoutingWorker.scala:366]: I spent a bit of time wrapping my head around this only to realize that the original code had a seeming bug in it. The types do not match. In some cases it returns a `RoutingResponse` and in others it returns an `Option[RoutingResponse]`

Here is my stab at stabilizing this method:

```
    if (modesToExclude.isEmpty) {
      r5.calcRoute(request)
    } else {
      val filteredStreetVehicles = request.streetVehicles.filterNot(it => modesToExclude.contains(it.mode))
      val r5ResponseOption = if (filteredStreetVehicles.isEmpty) {
        None
      } else {
        Some(r5.calcRoute(request.copy(streetVehicles = filteredStreetVehicles)))
      }

      val definedResponses = responses.flatten
      definedResponses.headOption.map{
        _.copy(definedResponses.flatMap(_.itineraries).toSeq ++
            r5ResponseOption.map(_.itineraries).getOrElse(Seq.empty))
      }.orElse(r5ResponseOption)
      .getOrElse(/*What to do here - this is the case where there is NO response*/)
```

Note that there is some undefined behavior because it previously returned an Option where the None was not being handled.

Comment (crixal) [reply src/main/scala/beam/router/RoutingWorker.scala:366]: Justin, thanks for your fix. R5 should be called for full request in the last case.

**Comment by crixal** [comment]
test!

**Review by JustinPihony — APPROVED** [review]

**Comment by JustinPihony** [comment]
@nikolayilyin Are your items still needed to be blockers or can you approve? Also, is this ready for merge @crixal ?

**Review by nikolayilyin — APPROVED** [review]

**Comment by JustinPihony** [comment]
@crixal Ready to merge?

**Review by dimaopen — CHANGES_REQUESTED** [review]
We cannot merge it into develop, because this branch contains the SFBay-prod branch in it.

**Comment by JustinPihony** [comment]
@dimaopen Should we work on removing those parts so we can merge? @crixal ?

**Comment by dimaopen** [comment]
> Should we work on removing those parts so we can merge?

@JustinPihony 
This PR can be closed. Because I've already extracted the actual code changes into another PR: https://github.com/LBNL-UCB-STI/beam/pull/3307

**Pull request closed without merging**
